src/minichain/consensus.py

```
@@ -0,0 +1,101 @@
+"""Consensus and Proof-of-Work mining primitives."""
+
+from __future__ import annotations
+
+from dataclasses import replace
+from threading import Event
+from typing import Sequence
```

Ruff (0.15.2)

[warning] Import from `collections.abc` instead: `Sequence`

Import from `collections.abc`

(UP035)